lib/cli-engine.js

```diff
@@ -694,6 +690,8 @@ CLIEngine.prototype = {
                     descriptor.meta.hashOfConfig = hashOfConfig;
                     descriptor.meta.results = res;
                 }
+            } else {
```

Comment:
Not sure I follow - which first `if` block? I thought we were destroying the cache if it wasn't called with the `--cache` flag